javaslang-benchmark/src/test/java/javaslang/MemoryUsage.java

```diff
@@ -0,0 +1,38 @@
+package javaslang;
+
+import javaslang.collection.*;
+
+import java.text.DecimalFormat;
+import java.util.Comparator;
+
+import static com.carrotsearch.sizeof.RamUsageEstimator.*;
+
+public class MemoryUsage {
+    private static final DecimalFormat RATIO_FORMAT = new DecimalFormat("#0.0");
+    private static TreeMultimap<Integer, String> memoryUsages = TreeMultimap.withSeq().empty(Comparator.reverseOrder());
+
+    /** Calculate the occupied memory of different internals */
+    static void printAndReset() {
+        for (int size : memoryUsages.keySet()) {
+            System.out.println(String.format("\nfor %d elements", size));
+            for (String usages : memoryUsages.get(size).get()) {
+                System.out.println("\t" + usages);
+            }
+        }
+
+        memoryUsages = memoryUsages.dropWhile(v -> true); // reset
```

Comment:
what happens if we `take(0)`?
_(We once had `clear()` but removed it per user request because it looked mutable and is not known from Scala.)_